ha/alert/iem.py

```diff
@@ -37,51 +41,66 @@ def __init__(self):
         """
         Init IEM generator
         """
-        self._execute = SimpleCommand()
         with open(IEM_SCHEMA, 'r') as iem_schema_file:
             self.iem_alert_data = json.load(iem_schema_file)
 
+    def _get_node_id(self, node: str) -> str:
+        '''
+        Temporary routine to get node id
+        '''
+
+        conf_store = ConfigManager.get_confstore()
+        key_val  = conf_store.get(f"{PVTFQDN_TO_NODEID_KEY}/{node}")
+        _, node_id = key_val.popitem()
+        return node_id
+
     def generate_iem(self, node: str, module: str, event_type: str) -> None:
         '''
-           Forms an IEC based on diffrent values such as module:<node/resource>
+           Creates an IEM based on diffrent values such as module:<node/resource>
            and event_type<lost/member for a node scenario>
-           IEC code:
-           IEC:{severity}{source}{component}{module_id}{event_id}:{desciption}
-           severity of the event.
-           source (Hardware or Software) of the event
-           component who is generating an IEM
+
+           severity : severity of the event.
+           source : source (Hardware or Software) of the event
+           component : component who is generating an IEM
            event_id: unique identification of the event (like node lost or node now became member)
            module: sub-component of the module who generated an IEM
-           Ex:
-           IEC:WS0080010001: node is down(node lost)
-           IEC:IS0080010002: node is up(node is now member)
 
            Required parameters
            node : Node name
            module : Module type (ex 'node' or 'resource' )
            event_type : Type of event based on module ( ex 'member' / 'lost' when module is 'node' )
         '''
+        module_type = self.iem_alert_data.get(module)
+        severity = module_type.get('severity').get(event_type)
+        source = module_type.get('source')
+        component = module_type.get('component')
+        module = module_type.get('module')
+        event_id = module_type.get('event').get(event_type).get('ID')
+        desc = module_type.get('event').get(event_type).get('desc')
+        description = re.sub("\$host", node, desc)
+        description = re.sub("\$status", event_type, description)
+        Log.info(f'Sending an IEM for: {component}-{module} to IEM framework. Description:  {description}')
+
+        try:
+            EventMessage.init(component, source)
```

Comment:
Can we initialize this in the constructor of this class?

Reply:
if it is static method then may not need it 
but goot to have in __init__

Reply:
Moved to init